### PR TITLE
Interpret a version from a ref

### PR DESCRIPTION
**Pull request by huntc**
Use the ref field of a web hook to determine a version number. The web hook conveys which branch should be updated.

Fixed up a failing test also.

### CODE REVIEW

File: app/controllers/Application.scala [hunk 1]
```diff
@@ -1,71 +1,82 @@
...
+          case None                    => None
+        }
       }
-    }
+      .flatMap(_.get(pathVersion))
   }
 }
 
 class Application @Inject() (
   @Named("ConductRDocRenderer") conductrDocRenderer: ActorRef,
```
Comment (markusjura) [inline app/controllers/Application.scala]: Why do we need this `ConductRDocRenderer`?

File: app/controllers/Application.scala [hunk 2]
```diff
@@ -73,31 +84,38 @@ class Application @Inject() (
...
+  private val docRenderers = Map(
+    "conductr" -> Map(
+      "" -> conductrDocRenderer,
+      "1.0.x" -> conductrDocRenderer10,
+      "1.1.x" -> conductrDocRenderer11)
+  )
 
   private val secret = new SecretKeySpec(settings.play.crypto.secret.getBytes, MacAlgorithm)
 
+  private val webhookRef = (JsPath \ "ref").read[String]
```
Comment (markusjura) [inline app/controllers/Application.scala]: Would put `secret` and `webhookRef` close to the point where it is used. In this case above the `update` method. This is more readable when looking the first time on it.

File: app/doc/DocVersions.scala [hunk 3]
```diff
@@ -1,5 +0,0 @@
-package doc
-
-object DocVersions {
-  val Latest = "1.0.x"
```
Comment (markusjura) [inline app/doc/DocVersions.scala:4]: Why do we not encapsulate the versions in an singleton object as before. Currently we are using the versions in multiple places:
- https://github.com/huntc/project-doc/blob/more-versions/app/modules/ConductRDocRendererModule.scala#L36
- https://github.com/huntc/project-doc/blob/more-versions/app/controllers/Application.scala#L90

Therefor we should encapsulate the strings to constants inside of a singleton object.

**Comment by markusjura** [comment]
LGTM (beside comments which are only style related). Also tested it locally and it is working fine.

File: app/controllers/Application.scala (hunk 1 above)
Comment (huntc) [inline app/controllers/Application.scala]: Good question. We no longer need to do this. The mapping of versions to renderers should be localised to the Application. I will change this.

File: app/controllers/Application.scala (hunk 2 above)
Comment (huntc) [inline app/controllers/Application.scala]: Will do.

File: app/doc/DocVersions.scala (hunk 3 above)
Comment (huntc) [inline app/doc/DocVersions.scala:4]: The problem with the previous approach was that it was tied to ConductR. Project-doc should remain agnostic with exception to the module declarations and how they relate to URI paths.

**Comment by huntc** [comment]
Great. I'll create a test for the webhook notification via the controller today in order to finish it off. Thanks.

File: app/doc/DocVersions.scala (hunk 3 above)
Comment (markusjura) [inline app/doc/DocVersions.scala:4]: That's fair.

**Comment by markusjura** [comment]
LGTM

**Pull request merged**